zerver/lib/home.py

```diff
@@ -35,6 +35,16 @@ class UserPermissionInfo:
     show_webathena: bool
 
 
+def is_outdated_server() -> bool:
+    # Returns (needs_upgrade)
+    server_upgrade_nag_deadline = settings.SERVER_UPGRADE_NAG_DEADLINE
+    last_server_upgrade_time = settings.LAST_SERVER_UPGRADE_TIME
+
+    if last_server_upgrade_time - server_upgrade_nag_deadline >= 31:
```

Comment:
Where does this 31 come from?

I think you need to think more about what logic makes sense here.

Reply:
Because the server upgrade nudge is suppose to be shown when it's 1-month past the server upgrade deadline(based on the info provided on the issue), I'm checking to see if the difference between the two variables is equal/greater than 31(1 month). I don't know if that make sense. I will try to see if there's a better way of doing that though.